### PR TITLE
DOC: Repair doc/source/merging.rst flake8 issues

### LINKED ISSUE #24179: DOC: Fix flake8 issues in doc/source/merging.rst
We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

The first step of this issue would be edit `setup.cfg` in the pandas home, and in the `flake8-rst` section, remove from the `exclude` list the file `doc/source/merging.rst`

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version `0.7.0` or higher):

```
$ flake8-rst doc/source/merging.rst 
doc/source/merging.rst:42:24: E127 continuation line over-indented for visual indent
doc/source/merging.rst:48:25: E127 continuation line over-indented for visual indent
doc/source/merging.rst:54:24: E127 continuation line over-indented for visual indent
doc/source/merging.rst:383:47: E231 missing whitespace after ','
doc/source/merging.rst:383:54: E231 missing whitespace after ','
doc/source/merging.rst:440:20: E128 continuation line under-indented for visual indent
doc/source/merging.rst:441:20: E128 continuation line under-indented for visual indent
doc/source/merging.rst:729:28: E203 whitespace before ':'
doc/source/merging.rst:729:33: E231 missing whitespace after ','
doc/source/merging.rst:729:41: E203 whitespace before ':'
doc/source/merging.rst:731:29: E203 whitespace before ':'
doc/source/merging.rst:731:34: E231 missing whitespace after ','
doc/source/merging.rst:731:36: E231 missing whitespace after ','
doc/source/merging.rst:731:48: E231 missing whitespace after ','
doc/source/merging.rst:731:50: E231 missing whitespace after ','
doc/source/merging.rst:804:50: E231 missing whitespace after ':'
doc/source/merging.rst:805:41: E231 missing whitespace after ','
doc/source/merging.rst:805:53: E231 missing whitespace after ':'
doc/source/merging.rst:860:83: E501 line too long (81 > 79 characters)
doc/source/merging.rst:869:9: E126 continuation line over-indented for hanging indent
doc/source/merging.rst:906:25: E127 continuation line over-indented for visual indent
doc/source/merging.rst:910:26: E127 continuation line over-indented for visual indent
doc/source/merging.rst:1002:26: E127 continuation line over-indented for visual indent
doc/source/merging.rst:1041:23: E128 continuation line under-indented for visual indent
doc/source/merging.rst:1042:22: E128 continuation line under-indented for visual indent
doc/source/merging.rst:1137:29: E203 whitespace before ':'
doc/source/merging.rst:1142:36: E226 missing whitespace around arithmetic operator
doc/source/merging.rst:1157:25: E127 continuation line over-indented for visual indent
doc/source/merging.rst:1164:26: E127 continuation line over-indented for visual indent
doc/source/merging.rst:1167:63: E231 missing whitespace after ','
doc/source/merging.rst:1167:67: E231 missing whitespace after ','
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from `setup.cfg`. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.

### PR DESCRIPTION
**Pull request by addisonlynch**
Repairs flake8 issues, removes exclusion from ``setup.cfg``.

- [x] closes #24179 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/merging.rst [hunk 1]
```diff
@@ -857,19 +857,18 @@ The left frame.
...
 .. ipython:: python
 
-   right = pd.DataFrame({
-        'X': pd.Series(['foo', 'bar'],
-                       dtype=CategoricalDtype(['foo', 'bar'])),
-        'Z': [1, 2]
-   })
+   right = pd.DataFrame({'X': pd.Series(['foo', 'bar'],
+                        dtype=CategoricalDtype(['foo', 'bar'])),
+                        'Z': [1, 2]})
```
Comment (datapythonista) [inline doc/source/merging.rst]: not sure if this is pep8 correct, but it's misleading. The `dtype` seems to belong to `DataFrame` and not `Series`.

Comment (addisonlynch) [reply doc/source/merging.rst]: Oops...bumped ``dtype`` back in line with the ``Series`` open paren.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @addisonlynch

**Comment by jreback** [comment]
thanks

**Pull request merged**